src/cmd-buildextend-installer

```diff
@@ -137,18 +133,27 @@ def mkinitrd_pipe(tmproot, destf, compress=True):
         offset = destf.tell()
         if offset % 4:
             destf.write(b'\0' * (4 - offset % 4))
-    findproc = subprocess.Popen(['find', '.', '-mindepth', '1', '-print0'],
-                                cwd=tmproot, stdout=subprocess.PIPE)
+    files = subprocess.check_output(['find', '.', '-mindepth', '1', '-print0'],
+                                cwd=tmproot)
+    file_list = files.split(b'\0')
+    # If there's a root.squashfs, it _must_ be the first file in the cpio
+    # archive, since the dracut 20live module assumes its contents are at
+    # a fixed offset in the archive.
```

Comment:
I feel a little uneasy about assuming a fixed offset here. Though one thing that saves us is that we can assume a binding between the initrd image and the rootfs image. IOW, we could technically change this in a future release as long as we ensure that the hardcoded offset in f-c-c is adjusted accordingly. Do you agree with that assessment?

One sanity-check we can do in this script is seek to offset 124 and check for the squashfs magic. That way any deviation is detected at build time.

Reply:
Yup, I agree.  Great idea about adding a cross-check.